src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -173,6 +174,36 @@ public function loadClass($class)
                 @trigger_error(sprintf('The %s class is considered final%s. It may change without further notice as of its next major version. You should not extend it from %s.', $parent, self::$final[$parent], $name), E_USER_DEPRECATED);
             }
 
+            // Only check final methods if the class has a parent
+            if ($parent) {
+                foreach ($refl->getMethods(\ReflectionMethod::IS_PUBLIC | \ReflectionMethod::IS_PROTECTED) as $method) {
+                    try {
+                        $prototype = $method->getPrototype();
```

Comment:
I discovered it today, it is fetching the parent method, which is what we need, we could do that manually but we would have to support several edge cases (for instance a method of the parent of the parent extended).